api/types/src/bytecode.rs

```diff
@@ -46,6 +46,16 @@ pub trait Bytecode {
 
     fn function_is_view(&self, name: &IdentStr) -> bool;
 
+    fn struct_is_event(&self, name: &IdentStr) -> bool {
+        match self.metadata() {
+            Some(m) => match m.struct_attributes.get(name.as_str()) {
+                Some(attrs) => attrs.iter().any(|attr| attr.is_event()),
+                None => false,
+            },
+            None => false,
+        }
```

Comment:
You could do (probably even fancier with and_then)
```suggestion
        if let Some(m) = self.metadata() {
            if let Some(attrs) = m.struct_attributes.get(name.as_str()) {
                return attrs.iter().any(|attr| attr.is_event()),
            }
        }
        
        false
```